ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapContainerFragment.kt

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.home.mapcontainer
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.annotation.StringRes
+import androidx.appcompat.app.AlertDialog
+import androidx.navigation.fragment.NavHostFragment
+import com.google.android.ground.R
+import com.google.android.ground.databinding.MapContainerFragBinding
+import com.google.android.ground.model.Survey
+import com.google.android.ground.model.geometry.Point
+import com.google.android.ground.model.locationofinterest.LocationOfInterest
+import com.google.android.ground.model.locationofinterest.LocationOfInterestType
+import com.google.android.ground.repository.MapsRepository
+import com.google.android.ground.rx.*
+import com.google.android.ground.system.PermissionDeniedException
+import com.google.android.ground.system.SettingsChangeRequestCanceled
+import com.google.android.ground.ui.common.AbstractMapViewerFragment
+import com.google.android.ground.ui.home.BottomSheetState
+import com.google.android.ground.ui.home.HomeScreenFragmentDirections
+import com.google.android.ground.ui.home.HomeScreenViewModel
+import com.google.android.ground.ui.map.MapFragment
+import com.google.android.ground.ui.map.MapLocationOfInterest
+import com.google.android.ground.ui.map.MapType
+import com.google.common.collect.ImmutableList
+import com.uber.autodispose.ObservableSubscribeProxy
+import dagger.hilt.android.AndroidEntryPoint
+import java8.util.Optional
+import javax.inject.Inject
+import kotlin.math.max
+import timber.log.Timber
+
+/** Main app view, displaying the map and related controls (center cross-hairs, add button, etc). */
+@AndroidEntryPoint
+class MapContainerFragment : AbstractMapViewerFragment() {
+  @Inject lateinit var mapsRepository: MapsRepository
```

Comment:
Is it possible to move injected dependency to constructor?

Reply:
Done!